### DISCUSSION TITLE
Are we going to add support for Coudflare's AI Gateway?

### DISCUSSION THREAD

**Discussion by Removable** [betalgo/openai#498] [Ideas]
Here's the introduction from the [document page](https://developers.cloudflare.com/ai-gateway/) of the AI Gateway:
> Cloudflare’s AI Gateway allows you to gain visibility and control over your AI apps. By connecting your apps to AI Gateway, you can gather insights on how people are using your application with analytics and logging and then control how your application scales with features such as caching, rate limiting, as well as request retries, model fallback, and more. Better yet - it only takes one line of code to get started.

Is there anyone who is already trying to add support for it?

**Comment by kayhantolga** [comment]
I haven't decided yet, but I think we should keep third-party providers separate from the main repository. I am considering modifying the library to allow other providers to be added manually by the user, or to include these changes into a utilities/extension library.

**Reply by Removable** [reply]
Alright, I agree!
